vision_project/vision_models/imageloader.py

```diff
@@ -201,30 +254,75 @@ def create_dataset(self):
         )
         return dataset
 
-    def load_data(self):
+    def load_data(self, split):
         """
         Load and prepare the data for training and validation.
 
         Returns:
             tuple: Training dataset and validation dataset.
             Each dataset will have image and label tensors.
         """
+        # # Create the dataset
+        # dataset = self.create_dataset()
+        # print("Dataset is created, setting batch size")
+
+        # if self.batch_size:
+        #     print("Batching dataset to :", self.batch_size)
+        #     dataset = dataset.batch(self.batch_size)
+            
+        # else:
+        #     print("Batching dataset to default defined in constants:", constants.BATCH_SIZE)
+        #     dataset = dataset.batch(constants.BATCH_SIZE)
+
+        # print("Dataset created, you can now iterate over the dataset")
+
+        # return dataset
+        
+        # Filter the label dataframe based on the split
+        if split == 'train':
+            split_ids = self.train_ids
+        elif split == 'val':
+            split_ids = self.val_ids
+        elif split == 'test':
+            split_ids = self.test_ids
+        else:
+            raise ValueError(f"Unknown split: {split}")
+
+        # Filter the label dataframe based on the split_ids
+        self.label_df = self.label_df[self.label_df['study_id'].isin(split_ids)]
+
         # Create the dataset
         dataset = self.create_dataset()
-        print("Dataset is created, setting batch size")
+        print(f"Dataset for {split} is created, setting batch size")
 
         if self.batch_size:
-            print("Batching dataset to :", self.batch_size)
+            print(f"Batching dataset to: {self.batch_size}")
             dataset = dataset.batch(self.batch_size)
-            
         else:
-            print("Batching dataset to default defined in constants:", constants.BATCH_SIZE)
+            print(f"Batching dataset to default defined in constants: {constants.BATCH_SIZE}")
             dataset = dataset.batch(constants.BATCH_SIZE)
 
-        print("Dataset created, you can now iterate over the dataset")
+        print(f"Dataset for {split} created, you can now iterate over the dataset")
 
         return dataset
 
+    def analyze_split(self):
+        """
+        Analyze the split of the dataset.
+        """
+        self.label_df['split'] = pd.cut(
```

Comment:
Should this be self.label_coordinates_csv ? That is the file we're using to generate labels and features for Stage 1 model.